sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/interceptor/SasTokenCredentialInterceptor.java

```diff
@@ -25,9 +26,23 @@ public SasTokenCredentialInterceptor(SasTokenCredential credential) {
     @Override
     public Response intercept(Chain chain) throws IOException {
         HttpUrl requestURL = chain.request().url();
-        String delimiter = (requestURL.query() != null && requestURL.query() != "") ? "&" : "?";
-        requestURL.toString();
-        String newURL = requestURL.toString() + delimiter + this.credential.getSasToken();
+
+        String encodedQuery = requestURL.encodedQuery();
+        if (!CoreUtils.isNullOrEmpty(encodedQuery)) {
+            encodedQuery += "&";
+        }
+
+        String sasToken = this.credential.getSasToken();
+        // SAS token is already encoded so its safe to append it to the encoded query from source request.
+        encodedQuery += sasToken.startsWith("?")
+            ? sasToken.substring(1)
+            : sasToken;
+
+        HttpUrl newURL = requestURL
+            .newBuilder()
+            .query(encodedQuery)
```

Comment:
```suggestion
            .encodedQuery(encodedQuery)
```
Otherwise we are re-encoding the query string.